### PR TITLE
ITS: Optionally set BxByBz propagation in fitting

### PR DESCRIPTION
**Pull request by mconcas**

### CODE REVIEW

File: Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx [hunk 1]
```diff
@@ -821,8 +821,14 @@ bool TrackerTraits::fitTrack(TrackITSExt& track, int start, int end, int step, f
...
-    if (!propInstance->propagateToX(track, trackingHit.xTrackingFrame, getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
-      return false;
+    if (useBxByBz) {
+      if (!propInstance->PropagateToXBxByBz(track, trackingHit.xTrackingFrame, o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
+        return false;
+      };
+    } else {
+      if (!propInstance->propagateToX(track, trackingHit.xTrackingFrame, getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
+        return false;
+      };
```
Comment (shahor02) [inline Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx]: Actually, there is this method which takes the switch:
https://github.com/AliceO2Group/AliceO2/blob/82f5a7fefde6c60de00fc95e3c370d96d95698e0/Detectors/Base/include/DetectorsBase/Propagator.h#L92

Comment (mconcas) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx]: thanks

File: Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx [hunk 2]
```diff
@@ -821,9 +821,9 @@ bool TrackerTraits::fitTrack(TrackITSExt& track, int start, int end, int step, f
       return false;
     }
 
-    if (!propInstance->propagateToX(track, trackingHit.xTrackingFrame, getBz(), o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
+    if (!propInstance->propagateTo(track, trackingHit.xTrackingFrame, useBxByBz, o2::base::PropagatorImpl<float>::MAX_SIN_PHI, o2::base::PropagatorImpl<float>::MAX_STEP, mCorrType)) {
```
Comment (shahor02) [inline Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx]: Actually, should be `!useBxByBz` since `true` in this method means use BZ only.

Comment (mconcas) [reply Detectors/ITSMFT/ITS/tracking/src/TrackerTraits.cxx]: right, test is now running, tomorrow I'll remove the WIP

**Comment by mconcas** [comment]
bzVSbxyz


Run a small test on 5 CTFs of 544490 (PbPb 43kHz), pt spectrum seems reasonably similar (red: Bz-only, Blue: Bxyz).
Timing seems even comparable:

```
bz only: ITS CA-Tracker total timing: Cpu: 6.796e+03 Real: 6.794e+03 s in 105 slots
bxbybz: ITS CA-Tracker total timing: Cpu: 6.441e+03 Real: 6.439e+03 s in 105 slots
```

**Comment by shahor02** [comment]
Thanks, please keep it on hold, will reprocess some of my K0 tests.

**Comment by shahor02** [comment]
Thanks! Checked K0s for the same pp chunks with and w/o `ITSCATrackerParam.useBxByBz=true`.
As expected, it does not make any difference.
k0_bzvs3d
So, I would close this ...

**Comment by mconcas** [comment]
> Thanks! Checked K0s for the same pp chunks with and w/o `ITSCATrackerParam.useBxByBz=true`. As expected, it does not make any difference. k0_bzvs3d So, I would close this ...

Fine with me. We can always resume this if it becomes necessary in the future.

**Pull request closed without merging**